clippy_lints/src/methods/unnecessary_iter_cloned.rs

```diff
@@ -34,6 +34,7 @@ pub fn check_for_loop_iter(
     expr: &'tcx Expr<'tcx>,
     method_name: Symbol,
     receiver: &'tcx Expr<'tcx>,
+    applicability: Applicability,
```

Comment:
Passing applicability to this function is difficult to make sense of. Does this sound right? If it is cloned (or to_owned) *before* `into_iter` is called, that's where we have issues.

```suggestion
    cloned_before_iter: bool,
```

Reply:
That's right: if there is a call to `clone`/`to_owned` before the call to `into_iter`, the suggestion to remove the `clone`/`to_owned` call could be incorrect, as could follow-on suggestions to remove uses of `.copied()`/`.cloned()` and `&`.

Should the effect of `cloned_before_iter` be the same: to determine the applicability of `check_for_loop_iter`'s suggestions?

Reply:
Yes, purely non-functional change to make the code easier to follow.

Reply:
Following the change, I felt like the comments should be adjusted. I can change them back, if preferred.